pkg/storage/storage.go

```diff
@@ -90,12 +90,14 @@ func getPlatformStorage() (imageregistryv1.ImageRegistryConfigStorage, error) {
 		return cfg, err
 	}
 
-	switch {
-	case installConfig.Platform.Libvirt != nil:
+	switch installConfig.Platform.Name() {
+	case "libvirt":
 		cfg.EmptyDir = &imageregistryv1.ImageRegistryConfigStorageEmptyDir{}
-	case installConfig.Platform.AWS != nil:
+	case "aws":
 		cfg.S3 = &imageregistryv1.ImageRegistryConfigStorageS3{}
-	case installConfig.Platform.OpenStack != nil:
+	case "azure":
+		cfg.EmptyDir = &imageregistryv1.ImageRegistryConfigStorageEmptyDir{}
+	case "openstack":
```

Comment:
I think this case statement was better w/ the field nil check since I don't know what assurances we have that the platform "name" won't change in the future, but the field should certainly not be nil when we're on a given platform.

that is, i'm not sure to what degree we can consider Platform.Name() to be a stable api.

if @abhinavdahiya or @sdodson or @wking says it's ok, we can, but otherwise i'd prefer to keep it as it was.